Sources/_FoundationInternals/String/BidirectionalCollection.swift

```diff
@@ -61,4 +61,72 @@ extension BidirectionalCollection {
         }
         return self[startOfNonTrimmedRange ... backIdx]
     }
+
+    // Equal to calling `index(&idx, offsetBy: -other.count)` with just one loop
+    func _index<S: BidirectionalCollection>(_ index: Index, backwardsOffsetByCountOf other: S) -> Index? {
+        var idx = index
+        var otherIdx = other.endIndex
+        while otherIdx > other.startIndex  {
+            guard idx > startIndex else {
+                // other.count > self.count: bail
+                return nil
+            }
+            other.formIndex(before: &otherIdx)
+            formIndex(before: &idx)
+        }
+        return idx
+    }
+
+    func _range<S: BidirectionalCollection>(of other: S, anchored: Bool = false, backwards: Bool = false) -> Range<Index>? where S.Element == Element, Element : Equatable {
```

Comment:
Can you add some tests for this version of `_range`?

Reply:
My bad for missing that. Will do